### PR TITLE
FM2-350: Add support for mapping visit types to the Encounter.type field

### PR DESCRIPTION
**Pull request by Medhavi-16**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-350

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Medhavi-16** [comment]
@ibacher I added mapping of visit types to Encounter.type field. PTAL

**Comment by ibacher** [comment]
@Medhavi-16 This looks good; however, I changed how the `ENCOUNTER_TYPE_SYSTEM_URI` is generated [in this commit](https://github.com/openmrs/openmrs-module-fhir2/commit/0aed612b4a7a0a79e535b5b19c9c9b31a25fcb16), which is what's causing the conflict. Would you be able to update this branch to be in line with that change?

**Comment by Medhavi-16** [comment]
> @Medhavi-16 This looks good; however, I changed how the `ENCOUNTER_TYPE_SYSTEM_URI` is generated [in this commit](https://github.com/openmrs/openmrs-module-fhir2/commit/0aed612b4a7a0a79e535b5b19c9c9b31a25fcb16), which is what's causing the conflict. Would you be able to update this branch to be in line with that change?

@ibacher Yes, I will update this branch according to that commit.

File: api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
 	        + "/condition-ver-status";
 	
 	public static final String CLINICAL_FINDINGS_SYSTEM_URI = HL7_FHIR_CODE_SYSTEM_PREFIX + "/clinical-findings";
-	
+
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java:53]:
```suggestion

```

**Review by ibacher — APPROVED** [review]

**Pull request merged**